test/vtdb_test.py

```diff
@@ -375,6 +398,38 @@ def test_tablet_restart_stream_execute(self):
       self.fail("Communication with shard0 replica failed with error %s" %
                 str(e))
 
+  def test_fail_mid_stream_execute(self):
+    """Tests for errors in the middle of a stream execute.
+
+    This is for testing app failures, not infrastructure failures. We simulate
+    an app failure by killing MySQL on the master tablet.
+
+    This test is inherently racy, as it tries to kill MySQL in the middle
```

Comment:
sleep(X) on the mysql side? I'm trying to interrupt a streaming query after at least the first QueryResult is returned. I'm not sure sleeping MySQL is any less racy.